x/gamm/pool-models/internal/cfmm_common/lp_test.go

```diff
@@ -0,0 +1,138 @@
+package cfmm_common_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/balancer"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/internal/cfmm_common"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/stableswap"
+	gammtypes "github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// a helper function used to multiply coins
+func mulCoins(coins sdk.Coins, multiplier sdk.Dec) sdk.Coins {
+	outCoins := sdk.Coins{}
+	for _, coin := range coins {
+		outCoin := sdk.NewCoin(coin.Denom, multiplier.MulInt(coin.Amount).TruncateInt())
+		if !outCoin.Amount.IsZero() {
+			outCoins = append(outCoins, outCoin)
+		}
+	}
+	return outCoins
+}
+
+func TestCalcExitPool(t *testing.T) {
+	var (
+		emptyContext = sdk.Context{}
+
+		// create these pools used for testing
+		twoAssetPool, err1 = stableswap.NewStableswapPool(
+			1,
+			stableswap.PoolParams{ExitFee: sdk.ZeroDec()},
+			sdk.NewCoins(sdk.NewInt64Coin("foo", 1000000000),
+				sdk.NewInt64Coin("bar", 1000000000)),
+			"",
+			time.Now(),
+		)
+		threeAssetPool, err2 = balancer.NewBalancerPool(
+			1,
+			balancer.PoolParams{SwapFee: sdk.ZeroDec(), ExitFee: sdk.ZeroDec()},
+			[]balancer.PoolAsset{
+				{Token: sdk.NewInt64Coin("foo", 2000000000), Weight: sdk.NewIntFromUint64(5)},
+				{Token: sdk.NewInt64Coin("bar", 3000000000), Weight: sdk.NewIntFromUint64(5)},
+				{Token: sdk.NewInt64Coin("baz", 4000000000), Weight: sdk.NewIntFromUint64(5)},
+			},
+			"",
+			time.Now(),
+		)
+		twoAssetPoolWithExitFee, err3 = stableswap.NewStableswapPool(
+			1,
+			stableswap.PoolParams{ExitFee: sdk.MustNewDecFromStr("0.0001")},
+			sdk.NewCoins(sdk.NewInt64Coin("foo", 1000000000),
+				sdk.NewInt64Coin("bar", 1000000000)),
+			"",
+			time.Now(),
+		)
+		threeAssetPoolWithExitFee, err4 = balancer.NewBalancerPool(
+			1,
+			balancer.PoolParams{SwapFee: sdk.ZeroDec(), ExitFee: sdk.MustNewDecFromStr("0.0002")},
+			[]balancer.PoolAsset{
+				{Token: sdk.NewInt64Coin("foo", 2000000000), Weight: sdk.NewIntFromUint64(5)},
+				{Token: sdk.NewInt64Coin("bar", 3000000000), Weight: sdk.NewIntFromUint64(5)},
+				{Token: sdk.NewInt64Coin("baz", 4000000000), Weight: sdk.NewIntFromUint64(5)},
+			},
+			"",
+			time.Now(),
+		)
+	)
+
+	// make sure there're no error creating those pools
+	require.NoError(t, err1)
```

Comment:
nit: 

Checking errors right after pool initialization might be preferable because we wouldn't have to wait until all pools are initialized to validate them all. Imagine, err1 had an error but we proceeded to initialize all pools redundantly to fail anyways.

In short, I'm pointing at something like the following:
```
twoAssetPool, err := stableswap.NewStableswapPool(
			1,
			stableswap.PoolParams{ExitFee: sdk.ZeroDec()},
			sdk.NewCoins(sdk.NewInt64Coin("foo", 1000000000),
				sdk.NewInt64Coin("bar", 1000000000)),
			"",
			time.Now(),
		)
require.NoError(t, err)

threeAssetPool, err = balancer.NewBalancerPool(
			1,
			balancer.PoolParams{SwapFee: sdk.ZeroDec(), ExitFee: sdk.ZeroDec()},
			[]balancer.PoolAsset{
				{Token: sdk.NewInt64Coin("foo", 2000000000), Weight: sdk.NewIntFromUint64(5)},
				{Token: sdk.NewInt64Coin("bar", 3000000000), Weight: sdk.NewIntFromUint64(5)},
				{Token: sdk.NewInt64Coin("baz", 4000000000), Weight: sdk.NewIntFromUint64(5)},
			},
			"",
			time.Now(),
		)
require.NoError(t, err)
```

Reply:
Tbf, I don't know why I did this, thanks for pointing out